benchmarks/units.py

```diff
@@ -86,3 +104,64 @@ def time_quantity_times_quantity():
 
 def time_quantity_ufunc_sin():
     np.sin(q2)
+
+
+class TimeQuantityOpSmallArray:
+    """
+    Operator benchmarks from https://github.com/astropy/astropy/issues/7546
+    for a small Numpy array.
+    """
+    def setup(self):
+        data = np.array([1., 2., 3.])
+        self.data = data * u.g
+        self.out = data.copy()
+
+        # A different but dimensionally compatible unit
+        self.data2 = 0.001 * data * u.kg
+
+    def time_quantity_square(self):
+        self.data ** 2
+
+    def time_quantity_np_square(self):
+        np.power(self.data, 2)
+
+    def time_quantity_np_square_out(self):
+        np.power(self.data, 2, out=self.out)
+
+    def time_quantity_sqrt(self):
+        self.data ** 0.5
+
+    def time_quantity_np_sqrt(self):
+        np.sqrt(self.data)
+
+    def time_quantity_np_sqrt_out(self):
+        np.sqrt(self.data, out=self.out)
+
+    def time_quantity_equal(self):
+        self.data == self.data2
+
+    def time_quantity_div(self):
+        # Since benchmark is PY3 only, this is always true divide.
+        self.data / self.data2
+
+    def time_quantity_mul(self):
+        self.data * self.data2
+
+    def time_quantity_sub(self):
+        self.data - self.data2
+
+    def time_quantity_add(self):
+        self.data + self.data2
+
+
+class TimeQuantityOpLargeArray(TimeQuantityOpSmallArray):
```

Comment:
@astrofrog , does subclassing like this work for `asv`?

Reply:
If the subclassing does work, that could also be done for the same-unit case.